### PR TITLE
Implement MaxSpanChunksSlice for Optimized Slasher

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Follow up to #8350, this PR implements the same logic but for max spans, ensuring we keep the same quality of tests.

**Which issues(s) does this PR fix?**

Part of #8331

### CODE REVIEW

File: beacon-chain/slasher/chunks.go [hunk 1]
```diff
@@ -264,6 +348,45 @@ func (m *MinSpanChunksSlice) Update(
...
+// Update a max span chunk for a validator index starting at a given start epoch, e_c, then updating
+// up to the current epoch according to the definition of max spans. If we need to continue updating
+// a next chunk, this function returns a boolean letting the caller know it should keep going.
+func (m *MaxSpanChunksSlice) Update(
+	chunkIdx uint64,
+	validatorIdx types.ValidatorIndex,
+	startEpoch,
+	currentEpoch,
+	newTargetEpoch types.Epoch,
+) (bool, error) {
```
Comment (rkapka) [inline beacon-chain/slasher/chunks.go:360]: We should really use named return values here and name the boolean `keepGoing`. Same for `MinSpanChunksSlice.Update` (actually the documentation of `MinSpanChunksSlice.Update` says "This function returns a boolean named keepGoing").

Comment (rkapka) [reply beacon-chain/slasher/chunks.go:360]: Naming aside, I don't understand why this function is designed to return the boolean. Why would anyone not want to keep going if more updating is needed? Shouldn't this function take care of it by itself? Maybe the exported function should accept a different set of parameters to have all the necessary info, and the current code should be placed in an unexported `updateInternal` function?

Comment (rkapka) [inline beacon-chain/slasher/chunks.go:360]: We should really use named return values here and name the boolean keepGoing. Same for MinSpanChunksSlice.Update (actually the documentation of MinSpanChunksSlice.Update says "This function returns a boolean named keepGoing").

**Review by rkapka — CHANGES_REQUESTED** [review]
Please use named return values in `Update`.

**Review by rkapka — APPROVED** [review]
Look great!

**Review by terencechain — APPROVED** [review]

**Pull request merged**